mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -5662,4 +5666,85 @@ protected override StringBuilder AppendNamespace (StringBuilder buf, TypeReferen
 	}
 }
 
+internal abstract class FlagsFormatter {
+	public abstract string Format(long c, Dictionary<long, string> values, string typename);
```

Comment:
hmm, really interesting approach ... I only went with `long` because that's what the existing implementation was using for the flags enumeration formatting. I'll take a look at refactoring this part